desktop/src/ui.rs

```diff
@@ -245,6 +308,32 @@ impl UiBackend for DesktopUiBackend {
     fn message(&self, message: &str) {
         message_box_ok("Ruffle", message, MessageBoxIcon::Info)
     }
+
+    fn display_file_dialog(&self, filters: Vec<FileFilter>) -> DialogResultFuture {
+        Box::pin(async move {
+            let mut dialog = AsyncFileDialog::new();
+
+            for filter in filters {
+                if std::env::consts::OS == "macos" {
+                    if let Some(mac_type) = filter.mac_type {
+                        let extensions: Vec<&str> = mac_type.split(';').collect();
+                        dialog = dialog.add_filter(&filter.description, &extensions);
+                    } else {
+                        let extensions: Vec<&str> = filter.extensions.split(';').collect();
+                        dialog = dialog.add_filter(&filter.description, &extensions);
+                    }
+                } else {
+                    let extensions: Vec<&str> = filter.extensions.split(';').collect();
+                    dialog = dialog.add_filter(&filter.description, &extensions);
```

Comment:
This isn't properly filtering for me on Windows: I get no results.

It looks like Flash expects input in a form like:
```
file_fr.browse([{description:"",extension:"*.JPG;*.jpg;*.gif;*.png"}]);
```
i.e. Windows extensions are preceded by wildcards and a dot (Mac extensions are alone).

`dialog.add_filter` expects extensions alone, without the star or dot.

Reply:
my bad, didn't check this properly.

Reply:
works now, thanks :)